### PR TITLE
created button for token generation 

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items
## Summary 
A shopping list is a set of items associated with a specific three-word token. We need to allow users to start new lists, so they can save the things they need to buy. To start a new list, we will need to do a few things:

1. Generate a new, unique token
2. Save the token to `localStorage`
3. Show the user the list view

Up to this point, the app has rendered the items in the list associated with the token `my test list`. Before we can allow users to make *new* lists, we have to stop showing them this test list.

## Acceptance criteria
- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user **doesn’t** already have a token:
- [x] A button in the `Home` component allows them to create a new list
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- The  `generateToken` function from the `shopping-list-utils` package should be used to generate the token. More information on that package [its documentation](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils). Importing it will look something like this:
``` js
import { generateToken } from '@the-collab-lab/shopping-list-utils';
```
- The `setListToken` function takes **one** argument: the new list token.
- Eventually the welcome/home page will look like [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png). It’s good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
- For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by mide358**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
This PR adds the `@the-collab-lab/shopping-list-utils` as a dependency to the project and also adds a button to the Home component which when clicked generates a three-word-token.

### Added `Create a new list` Button to Home component
A 'create a new list'  button was added to the home component. If clicked, it will generate a three-word-token that is stored in the localStorage using the setListToken function in App.jsx
If the user already has a token, the user is automatically redirected to the List view.

## Related Issue
Closes #3 

## Type of Changes

- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user doesn’t already have a token:

- [x] A button in the `Home` component allows them to create a new list
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.
if a user does already have a token:

- [x] They are automatically redirected to the `List` view.

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria
From your terminal, pull down this branch with `git pull origin ia-kl-new-shopping-list` and check that branch out with 
`git checkout  ia-kl-new-shopping-list`
Then run `npm i @the-collab-lab/shopping-list-utils`  to install the newly added dependency locally.
 Run npm start to launch the app.

Click on the `Create a new list` button to generate a token. (Note that the token is only visible in the local storage)
You will be redirected to the `List` page.

### CODE REVIEW

**Review by vicowiti — APPROVED** [review]

**Review by michhub-dev — APPROVED** [review]
Everything is working fine.

File: src/App.jsx [hunk 1]
```diff
@@ -51,7 +56,10 @@ export function App() {
 		<Router>
 			<Routes>
 				<Route path="/" element={<Layout />}>
-					<Route index element={<Home />} />
+					<Route
+						index
+						element={<Home onClick={handleClick} listToken={listToken} />}
+					/>
```
Comment (redapy) [inline src/App.jsx]: you can redirect the user directly from here, with the `<Navigate />` component, without passing the `listToken` to the home page. `element={listToken ? <Navigate to="/list" /> :  <Home onClick={handleClick} />}` 
- Note: This is a one way to fix the flash, maybe there are other solutions.

File: src/views/Home.jsx [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
-export function Home() {
+export function Home(props) {
+	const { listToken, onClick } = props;
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		if (listToken) {
+			navigate('/list');
+		}
+	}, [listToken, navigate]);
```
Comment (redapy) [inline src/views/Home.jsx]: If you use the `<Navigate />` component, you can just delete this.

**Review by redapy — CHANGES_REQUESTED** [review]
Excellent work team, the feature works as expected and AC is fulfilled.
I have just a tiny note when I try to navigate back to the home page (after the token is saved in my local storage) there is a quick flash (the 'generate token' button shows up in a sec). I left some comments that could be a solution!

**Comment by klezi10** [comment]
Thanks @redapy that's really good advice, using even less code!

**Review by redapy — APPROVED** [review]

**Pull request merged**